src/jit/decomposelongs.cpp

```diff
@@ -920,15 +925,24 @@ GenTree* DecomposeLongs::DecomposeShift(LIR::Use& use)
 
     GenTree* tree   = use.Def();
     GenTree* gtLong = tree->gtGetOp1();
+    GenTree* oldShiftOp = tree->gtGetOp2();
+
     genTreeOps oper = tree->OperGet();
+    genTreeOps shiftOper = oldShiftOp->OperGet();
+
```

Comment:
even more confusing is that below I see it referred to as "shiftWidthUse" which, IMO, should be shiftByUse.

---
In reply to: [78653812](https://github.com/dotnet/coreclr/pull/7169#discussion_r78653812) [](ancestors = 78653812)